### PR TITLE
Make Render Readme open a PR vs committing.

### PR DESCRIPTION
**Pull request by seabbs**
## Description

This PR is an attempt to partially address #447 as this is causing some development friction. Rather than continuing to try and commit README etc changes it instead makes the actions open a PR if the action has been triggered on main. 

This comes at a slight cost in that until the README action is merged the md README will be out of date with the Rmd README.

As part of this I have also removed the contributors workflow and put it within the render readme workflow.

### CODE REVIEW

**Comment by seabbs** [comment]
Tagged a few people who might have thoughts on if this actually helps or is also annoying!

**Review by adrian-lison — APPROVED** [review]
Thanks Sam!

I only looked at the changes but did not test it (how do you test this? :D)

But I like the approach. I would be up for giving this a try.

File: .github/workflows/render-readme.yaml [hunk 1]
```diff
@@ -28,17 +28,34 @@ jobs:
...
-          git add README.md man/figures/
-          git diff-index --quiet HEAD || git commit -m "Automatic readme update [ci skip]"
-          git push origin || echo "No changes to push"
+      - name: Check for README.Rmd changes
+        id: readme_changes
+        uses: dorny/paths-filter@v3
+        with:
+          filters: |
+            readme:
+              - 'README.Rmd'
```
Comment (Bisaloo) [inline .github/workflows/render-readme.yaml:49]: I don't believe this step is necessary as this is already handled in `peter-evans/create-pull-request` from what I can see.

**Review by Bisaloo — APPROVED** [review]
I agree this change leads to be better workflow.

File: .github/workflows/render-readme.yaml (hunk 1 above)
Comment (seabbs) [reply .github/workflows/render-readme.yaml]: Ah good point I've removed.

**Review by sbfnk — APPROVED** [review]
Looks great!

**Pull request merged**